src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -48,7 +46,15 @@ public ReadPileup(final Locatable loc, final List<PileupElement> pileup) {
      * Note: the current implementation of ReadPileup does not efficiently retrieve the stratified pileup
      */
     public ReadPileup(final Locatable loc, final Map<String, ReadPileup> stratifiedPileup) {
-        this(loc, stratifiedPileup.values().stream().flatMap(ReadPileup::getElementStream).collect(Collectors.toList()));
+        // NOTE: this bit of code is a potential performance hotspot for the HaplotypeCaller.
+        // This straightforward loop outperforms the equivalent streaming expression by over 2x.
```

Comment:
Maybe this will be improved too with #2321, but we should check if there is a regression there.